src/shim/util/connectionStringParser.ts

```diff
@@ -0,0 +1,51 @@
+import { ConnectionString, ConnectionStringKey } from "../types";
+import Constants = require("../../../src/declarations/constants");
+
+class ConnectionStringParser {
```

Comment:
This class is also available in OTel Exporter, why do we want to parse the connection string multiple times?, it would be better to not support very custom scenarios where this applies

Reply:
Given we're not supporting use cases that are this custom, I agree that this functionality seems extraneous. Removed this and the custom endpointUrl constructing.